easybuild/framework/easyblock.py

```diff
@@ -1042,12 +1044,15 @@ def make_module_dep(self, unload_info=None):
         self.log.debug("List of retained deps to load in generated module: %s", deps)
 
         # include load statements for retained dependencies
+        recursive_unload = self.cfg['recursive_module_unload']
+        if self.cfg['recursive_module_unload_depends_on']:
+            recursive_unload = 'depends_on'
```

Comment:
rather than introducing a magic value for the `recursive_unload` named argument of `load_module`, why not add a new named argument `use_depends_on`?

Reply:
Also, you should take the `build_option` into account here (rather than in `load_module`)?